contracts/Minter.sol

```diff
@@ -100,32 +104,62 @@ contract SPSMinter {
 
       emit Mint(pools[i].receiver, amount);
+
+      unchecked { ++i; }
     }
   }
 
   /**
    * @notice Add new pool, can be called by admin
    * @param newReceiver Address of the receiver
    * @param newAmount Amount of tokens per block
+   * @param newReductionBlocks Number of blocks between emission reduction
+   * @param newReductionPercentage Percentage to reduce emission
    */
-  function addPool(address newReceiver, uint256 newAmount) external onlyAdmin {
+  function addPool(address newReceiver, uint256 newAmount, uint256 newReductionBlocks, uint256 newReductionPercentage) external onlyAdmin {
     require(pools.length < poolsCap, 'SPSMinter: Pools cap reached');
     require(newAmount <= maxToPoolPerBlock, 'SPSMinter: Maximum amount per block reached');
-    pools.push(Pool(newReceiver, newAmount));
-    emit PoolAdded(newReceiver, newAmount);
+    pools.push(Pool(newReceiver, newAmount, newReductionBlocks, newReductionPercentage, block.number));
+    emit PoolAdded(newReceiver, newAmount, newReductionBlocks, newReductionPercentage, block.number);
   }
 
   /**
    * @notice Update pool, can be called by admin
    * @param index Index in the array of the pool
    * @param newReceiver Address of the receiver
    * @param newAmount Amount of tokens per block
+   * @param newReductionBlocks Number of blocks between emission reduction
+   * @param newReductionPercentage Percentage to reduce emission
    */
-  function updatePool(uint256 index, address newReceiver, uint256 newAmount) external onlyAdmin {
+  function updatePool(uint256 index, address newReceiver, uint256 newAmount, uint256 newReductionBlocks, uint256 newReductionPercentage) external onlyAdmin {
     require(newAmount <= maxToPoolPerBlock, 'SPSMinter: Maximum amount per block reached');
     mint();
-    pools[index] = Pool(newReceiver, newAmount);
-    emit PoolUpdated(index, newReceiver, newAmount);
+    pools[index] = Pool(newReceiver, newAmount, newReductionBlocks, newReductionPercentage, block.number);
+    emit PoolUpdated(index, newReceiver, newAmount, newReductionBlocks, newReductionPercentage, block.number);
+  }
+
+  /**
+   * @notice Update emissions for pools
+   * @param index Index in the array of the pool
+   * @param updateAll If true, all pools will be updated
+   */
+  function updateEmissions(uint256 index, bool updateAll) public {
+    if (updateAll){
+      uint256 length = pools.length;
+      for (uint256 i = 0; i < length;){
+        if (block.number - pools[i].lastUpdate > pools[i].reductionBlocks){
+          pools[i].amountPerBlock = pools[i].amountPerBlock / 100 * (100 - pools[i].reductionPercentage);
+          pools[i].lastUpdate = block.number;
+        }
+
+        unchecked { ++i; }
+      }
+    } else {
+      if (block.number - pools[index].lastUpdate > pools[index].reductionBlocks){
+        pools[index].amountPerBlock = pools[index].amountPerBlock / 10000 * (10000 - (pools[index].reductionPercentage * 100));
```

Comment:
Why the extra precision multiplier here but not in the other branch?

Reply:
I updated to basis points instead of percentage for higher precision: https://github.com/steem-monsters/sps-minter/commit/5f52f7dc2c53a1f5af54112801299f32f9499dc6